server/supernova/server/server.hpp

```diff
@@ -226,7 +226,11 @@ class nova_server : public asynchronous_log_thread,
 
 inline void run_scheduler_tick(void) {
     const int blocksize = sc_factory->world.mBufLength;
-    const int input_channels = sc_factory->world.mNumInputs;
+    // FIXME/TODO
+    // sc_factory and server end up with different values for number of input channels
+    // this was done in pr 4575 to fix PortAudio issue when requesting more channels than the harware provides
+    // consider making these values unified
```

Comment:
i would recommend "server may have fewer input channels than sc_factory"; if i were reading this fresh i would want to know that, because it rules out the possibility of certain bugs (i.e., i don't have to go and check whether it's possible for server to have _more_ input channels, and i also don't have to check whether that would even cause problems).

PR # and reason are not particularly relevant here, in my opinion, and can be found with git blame/log. personally i would leave them out, but it's up to you.

Reply:
That sounds good, I used your suggestion. However, do you think whether another comment needs to be added to prevent someone from just switching this back to `sc_factory->world.mNumInputs;`? The mistake won't be immediately apparent, unless they test it requesting higher number of channels than the hardware provides.

Sounds like a task for a unit test, but testing in server/cpp is uncharted territory for me.

Reply:
it looks good to me as is. i would hope anyone reviewing would be suspicious of that kind of change happening with a "FIXME" right above it :)